nvc/extract.go

```diff
@@ -0,0 +1,175 @@
+package nvc
+
+import (
+	"encoding/binary"
+	"errors"
+	"fmt"
+	"hash/fnv"
+	"io"
+	"os"
+	"strconv"
+)
+
+// TODO: maybe pick a better name for this type
+type compressionType uint32
+
+const (
+	uncompressed compressionType = 0
+	compressed   compressionType = 1
+	encrypted    compressionType = 3
+
+	nvcMagicBytes = "nvc1d\x00\x00\x00"
+)
+
+type nvcEntry struct {
+	Hash        uint64
+	Offset      uint32
+	RawLength   uint32 // Uncompressed length
+	Length      uint32 // Actual length in file
+	Compression compressionType
+}
+
+func Hash2String(hash uint64) string {
+	return fmt.Sprintf("%016x", hash)
+}
```

Comment:
Maybe we could do `type hash uint64` and then rename this function `func (h hash) String()` (to implement the `Stringer` interface)? Not really important though

Reply:
I'll add a TODO.